source/common/tls/utility.cc

```diff
@@ -227,9 +227,35 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     break;
   }
   case GEN_OTHERNAME:
-    str = general_name->d.otherName->value->value.asn1_string;
-    san.assign(reinterpret_cast<const char*>(ASN1_STRING_data(str)), ASN1_STRING_length(str));
-    break;
+    ASN1_TYPE* value = general_name->d.otherName->value;
+    if (value == nullptr) {
+      break;
+    }
+    switch (value->type) {
+    case V_ASN1_NULL:
+      break;
+    case V_ASN1_BOOLEAN:
+      san = value->value.boolean ? "true" : "false";
+      break;
+    case V_ASN1_ENUMERATED:
+      san = std::to_string(ASN1_ENUMERATED_get(value->value.enumerated));
```

Comment:
The docs say 

```
// WARNING: This function's return value cannot distinguish errors from -1.
// Use |ASN1_ENUMERATED_get_uint64| and |ASN1_ENUMERATED_get_int64| instead.
```

Reply:
Missed that one. Thanks for pointing it out. I looked into this a bit more and realized that getting it as an C int value limits the value you can get from a cert containing an INTEGER/ENUMERATED. I've modified it to use BIGNUM now. Also, made changes based on the comments below